Provider/Pool.php

```diff
@@ -45,8 +45,14 @@ public function __construct($context)
      */
     public function getProvider($name)
     {
+        if (!$name) {
+            throw new \InvalidArgumentException('Provider name cannot be empty, did you forget to call setProviderName() in your Media object?');
+        }
+        if (empty($this->providers)) {
+            throw new \RuntimeException('There are no providers configured yet');
+        }
         if (!isset($this->providers[$name])) {
-            throw new \RuntimeException(sprintf('unable to retrieve the provider named : `%s`', $name));
+            throw new \InvalidArgumentException(sprintf('Unable to retrieve the provider named "%s". Available providers are %s.', $name, '"'.implode('", "', $this->getProviderList()).'"'));
```

Comment:
I think the current syntax is better since these quotes you're suggesting to put arround ```%s``` are actually the opening and the closing quotes for the first and last array element in the list respectively.